rules/do-you-use-these-useful-react-hooks/rule.md

```diff
@@ -1,161 +1,259 @@
-        default:
-          throw new Error();
-      }
-    }
-    
-    function Counter() {
-      const [state, dispatch] = useReducer(counterReducer, { count: 0 });
-    
-      return (
-        <div>
-          <p>Count: {state.count}</p>
-          <button onClick={() => dispatch({ type: 'increment' })}>Increment</button>
-          <button onClick={() => dispatch({ type: 'decrement' })}>Decrement</button>
-        </div>
-      );
-    }
-    ```
-    
-    In the example above, useReducer is used to create a state variable called state and a dispatch function called dispatch. The counterReducer function is used as the reducer function. The initial state is an object with a count property set to 0. The Counter component uses the state.count value to render the current count, and the dispatch function to update the count by dispatching actions with the type property set to either 'increment' or 'decrement'. The reducer function updates the state based on the action dispatched by returning a new state object with the updated count value.
+## 1. **useState**: Managing Local State 🧠
+
+The `useState` hook lets you add state to functional components. Call `useState` at the top level of your component to declare one or more state variables.
+
+```jsx
+import { useState } from 'react';
+
+export default function Counter() {
+  const [count, setCount] = useState(0);
+
+  function handleClick() {
+    setCount(count + 1);
+  }
+
+  return (
+    <button onClick={handleClick}>
+      You pressed me {count} times
+    </button>
+  );
+}
+```
+**Figure: Using useState for a counter component**
+
+::: info
+***Naming Convention***:
+It's a common convention to name state variables using the pattern [**count**, **setCount**] with array destructuring.
+:::
+
+`useState` returns an array with exactly two items:
+
+1. The current state of this state variable, initially set to the initial state you provided.
+2. A function that lets you update its value.
+
+### ✅ Recommended Usage
+
+- **Updating Objects and Arrays**: Manage and adjust objects and arrays in the state. Remember, always create new references instead of mutating
+- **Avoiding Recreating the Initial State**: Ensure the initial state is set only once, avoiding recalculations in subsequent renders
+- **Resetting State with a Key**: Reset the component's state by altering its key
+- **Storing Information from Previous Renders**: On rare occasions, adjust state as a reaction to a rendering process
+
+### ⚠️ Pitfalls
+
+- **State Updates**: A change in state doesn't instantly reflect within the current executing code. It determines what `useState` will return in future renders
+- **Initializer Function**: When you pass a function to `useState`, it gets called only during the initialization phase
+- **State Updates with Functions**: When deriving new state values from the previous state, it's better to use an updater function. This ensures you're working with the most up-to-date state
```

Comment:
```suggestion
- **State Updates with Functions**: When deriving new state values from the previous state, it's better to use an updater function as an argument of the setter function instead of the new value i.e. `setObj(prev => { key: value ...prev }`. This ensures you're working with the most up-to-date state
```